File: java/core/src/test/java/com/github/pukkaone/mullet/html/EscapeXmlTest.java
/*
Copyright (c) 2011, Chin Huang
All rights reserved.

Redistribution and use in source and binary forms, with or without
modification, are permitted provided that the following conditions are met:

 * Redistributions of source code must retain the above copyright notice,
   this list of conditions and the following disclaimer.
 * Redistributions in binary form must reproduce the above copyright notice,
   this list of conditions and the following disclaimer in the documentation
   and/or other materials provided with the distribution.

THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
(INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
*/
package com.github.pukkaone.mullet.html;

import static org.junit.Assert.assertEquals;

import org.junit.Test;

/**
 * Tests escape-xml command.
 */
public class EscapeXmlTest extends TemplateTests {
        
    @Test
    public void should_disable_then_enable_escaping() 
        throws Exception
    {
        setModelValue("greeting", "<b>");
        
        Template template = loader.load("escape-xml.html");
        template.execute(data, writer);

        final String EXPECTED_OUTPUT =
                "<body>\n" +
                "  <p><b></p>\n" +
                "  <div>\n" +
                "    <p>&lt;b&gt;</p>\n" +
                "  </div>\n" +
                "</body>\n";
        assertEquals(EXPECTED_OUTPUT, writer.toString());
    }    
}

File: ruby/core/test/mullet/html/escape_xml_test.rb
require 'minitest/autorun'
require 'mullet/html/template_tests'

module Mullet; module HTML

  class EscapeXmlTest < MiniTest::Unit::TestCase
    include TemplateTests

    def test_should_disable_then_enable_escaping()
      set_variable('greeting', '<b>')

      template = @loader.load('escape-xml.html')
      template.execute(@data, @output)

      expected_output = <<EOF
<body>
  <p><b></p>
  <div>
    <p>&lt;b&gt;</p>
  </div>
</body>
EOF
      assert_equal expected_output, @output
    end
  end

end; end
